packages/ui/components/form-core/src/choice-group/CustomChoiceGroupMixin.js

```diff
@@ -0,0 +1,204 @@
+    static get properties() {
+      return {
+        allowCustomChoice: {
+          type: Boolean,
+          attribute: 'allow-custom-choice',
+        },
+        modelValue: { type: Object },
+      };
+    }
+
+    // @ts-ignore
+    get modelValue() {
+      return this.__getChoicesFrom(super.modelValue);
+    }
+
+    set modelValue(value) {
+      super.modelValue = value;
+
+      if (value === null || value === undefined || value === '') {
+        // @ts-ignore
+        this._customChoices = new Set();
+      } else if (this.allowCustomChoice) {
+        const old = this.modelValue;
+        // @ts-ignore
+        this._customChoices = new Set(normalizeArray(value));
+        this.requestUpdate('modelValue', old);
+      }
+    }
+
+    // @ts-ignore
+    get formattedValue() {
+      return this.__getChoicesFrom(super.formattedValue);
+    }
+
+    set formattedValue(value) {
+      super.formattedValue = value;
+
+      if (value === null || value === undefined) {
+        this._customChoices = new Set();
+      } else if (this.allowCustomChoice) {
+        const old = this.modelValue;
+        // Convert formattedValue to modelValue to store as custom choices, or fall back to the input value
+        this._customChoices = new Set(
+          normalizeArray(value).map(
+            val => this.formElements.find(el => el.formattedValue === val)?.modelValue || val,
+          ),
+        );
+        this.requestUpdate('modelValue', old);
+      }
+    }
+
+    // @ts-ignore
+    get serializedValue() {
+      return this.__getChoicesFrom(super.serializedValue);
+    }
+
+    set serializedValue(value) {
+      super.serializedValue = value;
+
+      if (value === null || value === undefined) {
+        this._customChoices = new Set();
+      } else if (this.allowCustomChoice) {
+        const old = this.modelValue;
+        // Convert serializedValue to modelValue to store as custom choices, or fall back to the input value
+        this._customChoices = new Set(
+          normalizeArray(value).map(
+            val => this.formElements.find(el => el.serializedValue === val)?.modelValue || val,
+          ),
+        );
+        this.requestUpdate('modelValue', old);
+      }
+    }
+
+    /**
+     * Custom elements are all missing elements that have no corresponding element, independent if enabled or not.
+     */
+    // @ts-ignore
+    get customChoices() {
+      if (!this.allowCustomChoice) {
+        return [];
+      }
+
+      const elems = this._getCheckedElements();
+
+      return Array.from(this._customChoices).filter(
+        choice => !elems.some(elem => elem.choiceValue === choice),
+      );
+    }
+
+    constructor() {
+      super();
+
+      this.allowCustomChoice = false;
+
+      /**
+       * @type {Set<unknown>}
+       * @protected
+       */
+      this._customChoices = new Set();
+
+      /**
+       * @private
+       */
+      this.__initialized = false;
```

Comment:
Is it possible to reuse `hasUpdated` instead? https://github.com/lit/lit/blob/main/packages/reactive-element/src/reactive-element.ts#L1489